src/test/java/io/candydoc/ddd/tactical_patterns/factory/FactoryRulesTest.java

```diff
@@ -0,0 +1,70 @@
+package io.candydoc.ddd.tactical_patterns.factory;
+
+import static io.candydoc.ddd.AssertDependency.assertThatDependency;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.candydoc.ddd.model.Dependency;
+import io.candydoc.ddd.model.DomainModel;
+import io.candydoc.ddd.model.FullName;
+import io.candydoc.ddd.model.PackageName;
+import io.candydoc.ddd.model.SimpleName;
+import io.candydoc.ddd.tactical_pattern.Factory;
+import io.candydoc.ddd.tactical_pattern.TacticalPattern;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class FactoryRulesTest {
+  static DomainModel factory;
+
+  @BeforeAll
+  static void setUp() {
+    factory =
+        Factory.builder()
+            .fullName(FullName.of("io.candydoc.sample.FactoryOne"))
+            .simpleName(SimpleName.of("FactoryOne"))
+            .packageName(PackageName.of("io.candydoc.sample"))
+            .dependency(
+                Dependency.builder()
+                    .name(FullName.of("io.candydoc.sample.ValueObjectOne"))
+                    .pattern(TacticalPattern.VALUE_OBJECT)
+                    .build())
+            .dependency(
+                Dependency.builder()
+                    .name(FullName.of("io.candydoc.sample.AggregateOne"))
+                    .pattern(TacticalPattern.AGGREGATE)
+                    .build())
+            .dependency(
+                Dependency.builder()
+                    .name(FullName.of("io.candydoc.sample.FactoryTwo"))
+                    .pattern(TacticalPattern.FACTORY)
+                    .build())
+            .build();
+  }
+
+  @Test
+  void is_a_factory() {
+    assertThat(factory.dddPattern()).isEqualTo(TacticalPattern.FACTORY);
+  }
+
+  @Test
+  void it_can_depend_on_value_object() {
+    Dependency valueObject =
+        Dependency.builder()
+            .name(FullName.of("io.candydoc.sample.ValueObjectOne"))
+            .pattern(TacticalPattern.VALUE_OBJECT)
+            .build();
+
+    assertThatDependency(valueObject).isAllowedIn(factory);
+  }
+
+  @Test
+  void it_can_depend_on_aggregate() {
+    Dependency aggregate =
+        Dependency.builder()
+            .name(FullName.of("io.candydoc.sample.AggregateOne"))
+            .pattern(TacticalPattern.AGGREGATE)
+            .build();
+
+    assertThatDependency(aggregate).isAllowedIn(factory);
+  }
```

Comment:
A factory can also create domain entities. And I guess factory can depends on other factory (the ALLOWED_DEPENDENCIES suggest it by the way) but I don't find any tests for that. Where's TDD :'( ?

Add two tests:
- it_can_depend_on_entities
- it_can_depend_on_other_factories

Reply:
You're right for the other factories. However, the domain entities don't exist on this branch, they only exist in the _feat/annotations/entity_ branch.